tensorflow/core/kernels/data/dataset_test_params.h

```diff
@@ -0,0 +1,211 @@
+#include "tensorflow/core/kernels/data/map_dataset_op.h"
+#include "tensorflow/core/kernels/data/name_utils.h"
+#include "tensorflow/core/kernels/data/range_dataset_op.h"
+
+namespace tensorflow {
+namespace data {
+
+typedef std::vector<
+    std::pair<string, tensorflow::FunctionDefHelper::AttrValueWrapper>>
+    AttributeVector;
+
+constexpr char kDefaultIteratorPrefix[] = "Iterator";
+
+// Creates a tensor with the specified dtype, shape, and value.
+template <typename T>
+static Tensor CreateTensor(const TensorShape& input_shape,
+                           const gtl::ArraySlice<T>& input_data) {
+  Tensor tensor(DataTypeToEnum<T>::value, input_shape);
+  test::FillValues<T>(&tensor, input_data);
+  return tensor;
+}
+
+// Creates a vector of tensors with the specified dtype, shape, and values.
+template <typename T>
+std::vector<Tensor> CreateTensors(
+    const TensorShape& shape, const std::vector<gtl::ArraySlice<T>>& values) {
+  std::vector<Tensor> result;
+  result.reserve(values.size());
+  for (auto& value : values) {
+    result.emplace_back(CreateTensor<T>(shape, value));
+  }
+  return result;
+}
+
+enum class DatasetParamsType {
+  Range,
+  Batch,
+  Map,
+};
+
+string ToString(DatasetParamsType type);
+
+class RangeDatasetParams;
+class BatchDatasetParams;
+class MapDatasetParams;
+
+class DatasetParams {
+ public:
+  DatasetParams(DataTypeVector output_dtypes,
+                std::vector<PartialTensorShape> output_shapes, string node_name,
+                DatasetParamsType type);
+
+  ~DatasetParams() {}
+
+  // Returns the dataset inputs as a TensorValue vector.
+  virtual Status MakeInputs(gtl::InlinedVector<TensorValue, 4>* inputs) = 0;
+
+  // Returns the dataset attributes as a vector.
+  virtual Status MakeAttributes(AttributeVector* attributes) const = 0;
+
+  // Checks if the tensor is a dataset variant tensor.
+  static bool IsDatasetTensor(const Tensor& tensor);
+
+  string node_name() const { return node_name_; }
+
+  DataTypeVector output_dtypes() const { return output_dtypes_; }
+
+  std::vector<PartialTensorShape> output_shapes() const {
+    return output_shapes_;
+  }
+
+  string iterator_prefix() const { return iterator_prefix_; }
+
+  DatasetParamsType type() const { return type_; }
+
+  std::vector<std::pair<std::shared_ptr<DatasetParams>, Tensor>>&
+  input_dataset_params() {
+    return input_dataset_params_group_;
+  }
+
+  virtual std::vector<FunctionDef> func_lib() { return {}; }
+
+ protected:
+  // Used to store all the input dataset parameters and the dataset tensors
+  // generated from the parameters.
+  std::vector<std::pair<std::shared_ptr<DatasetParams>, Tensor>>
+      input_dataset_params_group_;
+  DataTypeVector output_dtypes_;
+  std::vector<PartialTensorShape> output_shapes_;
+  string node_name_;
+  string iterator_prefix_ = kDefaultIteratorPrefix;
+  DatasetParamsType type_;
+};
+
+class BatchDatasetParams : public DatasetParams {
```

Comment:
I feel like this should be in `batch_dataset_op_test.h`, inlining the implementation of `MakeInputs` and `MakeAttributes` so that the header can be included from test .cc files that wish to use it

Putting all params objects for all dataset op kernels into this file does not scale.

Reply:
Great suggestion! Will change it as you suggest.